File: update_headers.py
import os
import glob

html_files = glob.glob('*.html')

for filepath in html_files:
    if not os.path.isfile(filepath): continue
    
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # 1. Update font sizes in header
    header_start = content.find('<header')
    header_end = content.find('</header>')
    if header_start != -1 and header_end != -1:
        header_end += len('</header>')
        header_chunk = content[header_start:header_end]
        header_chunk_updated = header_chunk.replace('text-lg', 'text-base')
        content = content[:header_start] + header_chunk_updated + content[header_end:]

    # 2. Reorder Mobile Nav First
    start_mobile = content.find('<!-- Service links -->')
    photo_mobile = content.find('<!-- Photography accordion -->')
    about_mobile = content.find('<a href="about.html"', photo_mobile)
    
    if start_mobile != -1 and photo_mobile != -1 and about_mobile != -1:
        top_links = content[start_mobile:photo_mobile].strip().split('\n')
        ev_link = next((l for l in top_links if 'event-marketing.html' in l), '')
        brand_link = next((l for l in top_links if 'brand-design.html' in l), '')
        web_link = next((l for l in top_links if 'web-development.html' in l), '')
        
        photo_chunk = content[photo_mobile:about_mobile]
        
        indent = "                    "
        if ev_link: indent = ev_link[:len(ev_link) - len(ev_link.lstrip())]
            
        new_mobile = (
            photo_chunk + 
            indent + "<!-- Service links -->\n" +
            brand_link + "\n" +
            web_link + "\n" +
            ev_link + "\n" +
            indent
        )
        content = content[:start_mobile] + new_mobile + content[about_mobile:]

    # 3. Reorder Desktop Nav Last
    start_desktop = content.find('<!-- Top-level service links -->')
    photo_desktop = content.find('<!-- Photography Dropdown -->')
    about_desktop = content.find('<a href="about.html"', photo_desktop)
    
    if start_desktop != -1 and photo_desktop != -1 and about_desktop != -1:
        top_links = content[start_desktop:photo_desktop].strip().split('\n')
        ev_link = next((l for l in top_links if 'event-marketing.html' in l), '')
        brand_link = next((l for l in top_links if 'brand-design.html' in l), '')
        web_link = next((l for l in top_links if 'web-development.html' in l), '')
        
        photo_chunk = content[photo_desktop:about_desktop]
        
        indent = "                    "
        if ev_link: indent = ev_link[:len(ev_link) - len(ev_link.lstrip())]
            
        new_desktop = (
            photo_chunk + 
            indent + "<!-- Top-level service links -->\n" +
            brand_link + "\n" +
            web_link + "\n" +
            ev_link + "\n" +
            indent
        )
        content = content[:start_desktop] + new_desktop + content[about_desktop:]

    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)

print('Updated DOM structure successfully!')

File: update_headers.js
import fs from 'fs';
import path from 'path';

const htmlFiles = fs.readdirSync('.').filter(file => file.endsWith('.html'));

for (const filepath of htmlFiles) {
    let content = fs.readFileSync(filepath, 'utf-8');

    // 1. Update font sizes in header
    const headerStart = content.indexOf('<header');
    let headerEnd = content.indexOf('</header>');
    if (headerStart !== -1 && headerEnd !== -1) {
        headerEnd += '</header>'.length;
        const headerChunk = content.substring(headerStart, headerEnd);
        const headerChunkUpdated = headerChunk.replace(/text-lg/g, 'text-base');
        content = content.substring(0, headerStart) + headerChunkUpdated + content.substring(headerEnd);
    }

    // 2. Reorder Mobile Nav First
    const startMobile = content.indexOf('<!-- Service links -->');
    const photoMobile = content.indexOf('<!-- Photography accordion -->');
    let aboutMobile = content.indexOf('<a href="about.html"', photoMobile);

    if (startMobile !== -1 && photoMobile !== -1 && aboutMobile !== -1) {
        const topLinks = content.substring(startMobile, photoMobile).trim().split('\n');
        const evLink = topLinks.find(l => l.includes('event-marketing.html')) || '';
        const brandLink = topLinks.find(l => l.includes('brand-design.html')) || '';
        const webLink = topLinks.find(l => l.includes('web-development.html')) || '';

        const photoChunk = content.substring(photoMobile, aboutMobile);

        let indent = "                    ";
        if (evLink) {
            indent = evLink.substring(0, evLink.length - evLink.trimLeft().length);
        }

        const newMobile = photoChunk +
            indent + "<!-- Service links -->\n" +
            brandLink + "\n" +
            webLink + "\n" +
            evLink + "\n" +
            indent;

        content = content.substring(0, startMobile) + newMobile + content.substring(aboutMobile);
    }

    // 3. Reorder Desktop Nav Last
    const startDesktop = content.indexOf('<!-- Top-level service links -->');
    const photoDesktop = content.indexOf('<!-- Photography Dropdown -->');
    const aboutDesktop = content.indexOf('<a href="about.html"', photoDesktop);

    if (startDesktop !== -1 && photoDesktop !== -1 && aboutDesktop !== -1) {
        const topLinks = content.substring(startDesktop, photoDesktop).trim().split('\n');
        const evLink = topLinks.find(l => l.includes('event-marketing.html')) || '';
        const brandLink = topLinks.find(l => l.includes('brand-design.html')) || '';
        const webLink = topLinks.find(l => l.includes('web-development.html')) || '';

        const photoChunk = content.substring(photoDesktop, aboutDesktop);

        let indent = "                    ";
        if (evLink) {
            indent = evLink.substring(0, evLink.length - evLink.trimLeft().length);
        }

        const newDesktop = photoChunk +
            indent + "<!-- Top-level service links -->\n" +
            brandLink + "\n" +
            webLink + "\n" +
            evLink + "\n" +
            indent;

        content = content.substring(0, startDesktop) + newDesktop + content.substring(aboutDesktop);
    }

    fs.writeFileSync(filepath, content, 'utf-8');
}

console.log('Updated DOM structure successfully!');
